### ISSUE TITLE
DNS Resolving problems on Azure container apps / requirements.txt pinning

### ISSUE THREAD

**Issue by anrub** [grindsa/acme2certifier#103]
I got DNS resolving errors validating the DNS challenge while hosting on Azure Container Apps. I traced it down to the `dns.resolver.query` call, which is deprecated in dns.resolver (at least there was a warning printed) .  I updated that call to  `dns.resolver.resolve`, which solved my problem.

The root cause is probably, that there are no versions pinned in the requirements.txt, so that in my rebuild I used some newer dns.resolver library.

Maybe we could add a pinned down requirements.txt to the desired versions?

**Comment by grindsa** [comment]
Do you have a log showing the issue?

**Comment by anrub** [comment]
In the build, I am using dnspython 2.3.0.
And the error while resolving the DNS challange looked something like this:

```
[Tue Feb 14 09:29:16.616259 2023] [wsgi:error] [pid 36:tid 140576464041728] [client 10.123.14.123:59978] txt_get() error: The DNS response does not contain an answer to the question: _acme-challenge.mydomain.de.k8se-apps.svc.cluster.local. IN TXT
[Tue Feb 14 09:29:16.613745 2023] [wsgi:error] [pid 36:tid 140576464041728] [client 10.123.14.123:59978] txt_get() info FQDN: _acme-challenge.mydomain.de
[Tue Feb 14 09:29:16.626555 2023] [wsgi:error] [pid 36:tid 140576464041728] [client 10.23.14.123:59978] 10.180.14.112 /acme/chall/ xyz {'data': {'type': 'dns-01', 'token': '- modified -', 'status': 'pending', 'url': 'https://xyz.westeurope.azurecontainerapps.io/acme/chall/ xyz }, 'header': {'Link': 'https://xyz.westeurope.azurecontainerapps.io/acme/authz/;rel="up"', 'Replay-Nonce': '- modified -'}, 'code': 200}
10.123.14.123 - - [14/Feb/2023:09:29:16 +0000] "POST /acme/chall/ xyz HTTP/1.1" 200 194
```

When using `dns.resolver.resolve` it resolves just normal. The deprecation notice was unfortunately swallowed in the logging (problem on my end).

I didn't track the error to the exact change in dnspython, but if I run the following snippet directly in a containerapp, I receive the same errors, that I have when the DNS validation happens.

The sample resolves a TXT with value "asdf", using resolve it works, using query it doesn't work.

```
import dns.resolver

fqdn = 'test.mydomain.de'
response = dns.resolver.resolve(fqdn, 'TXT')
for i in response:
    print(i.strings[0])

fqdn = 'test.mydomain.de'
response = dns.resolver.query(fqdn, 'TXT')
for i in response:
    print(i.strings[0])
``` 

```
b'asdf'

test.py:11: DeprecationWarning: please use dns.resolver.resolve() instead
  response = dns.resolver.query(fqdn, 'TXT')
Traceback (most recent call last):
  File "test.py", line 11, in <module>
    response = dns.resolver.query(fqdn, 'TXT')
  File "/usr/local/lib/python3.8/dist-packages/dns/resolver.py", line 1401, in query
    return resolve(
  File "/usr/local/lib/python3.8/dist-packages/dns/resolver.py", line 1368, in resolve
    return get_default_resolver().resolve(
  File "/usr/local/lib/python3.8/dist-packages/dns/resolver.py", line 1231, in resolve
    (answer, done) = resolution.query_result(response, None)
  File "/usr/local/lib/python3.8/dist-packages/dns/resolver.py", line 768, in query_result
    raise NoAnswer(response=answer.response)

dns.resolver.NoAnswer: The DNS response does not contain an answer to the question: test.mydomain.de.k8se-apps.svc.cluster.local. IN TXT
```

**Comment by grindsa** [comment]
For some reason I was not able to replicate your issue. Even when using dnspython 2.3.0 both 
`dns.resolver.resolve` and `dns.resolver.query` did work without issues. Are you using a special version of the module?  

Anyway, I change a few functions to use `dns.resolver.resolve` to get rid of the deprecation notice. Change is included in [v0.25.1](https://github.com/grindsa/acme2certifier/releases/tag/0.25.1) which got released last night. Feel free to give it a try and check if your issue got solved.
